EffectiveJava3Edition/src/main/java/stream/parallel_multithread/ParallelMersennePrimes.java

```diff
@@ -0,0 +1,24 @@
+package stream.parallel_multithread;
+
+import java.math.BigInteger;
+import java.util.stream.Stream;
+
+import static java.math.BigInteger.ONE;
+import static java.math.BigInteger.TWO;
+
+// Parallel stream-based program to generate the first 20 Mersenne primes - HANGS!!! (Page 222)
+public class ParallelMersennePrimes {
+	
+    public static void main(String[] args) {
+        primes().map(p -> TWO.pow(p.intValueExact()).subtract(ONE))
+                .parallel()
```

Comment:
Using parallel() method here may harm the machine since the calculations are huge!

Reply:
updated, line may cause problem commented out, though the reason I had it there was to show a non-usage of method :)